### PR TITLE
# 103 Added an interface and removed writer method

### PR DESCRIPTION
**Pull request by ammaratef45**
PR for #103 
created an interface with the required methods from XmlStreamWriter and implemented it in XmlFile class

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#109](https://github.com/artipie/rpm-adapter/issues/109) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/63d65bf1-8dd3-4eef-a1bc-46d43fd76370); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/rpm/meta/XmlWriter.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+package com.artipie.rpm.meta;
+
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * Interface for wrapping needed functionality of xmlstreamwriter.
+ * @since 0.7
+ */
+@SuppressWarnings("PMD.TooManyMethods")
+public interface XmlWriter {
```
Comment (Vatavuk) [inline src/main/java/com/artipie/rpm/meta/XmlWriter.java:33]: @ammaratef45 we don't need this interface, we can impelment `XMLStreamWriter`

File: src/main/java/com/artipie/rpm/meta/XmlFile.java [hunk 2]
```diff
@@ -58,8 +55,10 @@
  *  to XmlPackagesFile and renamed close by exploiting directly that information.
  *  Once it is done, use XmlPackagesFile in XmlFilelists, XmlOthers, XmlPrimary and keep
  *  using XmlFile in XmlRepomd.
+ * @checkstyle MethodsOrderCheck (500 lines)
  */
-final class XmlFile {
+@SuppressWarnings("PMD.TooManyMethods")
+final class XmlFile implements XmlWriter {
```
Comment (Vatavuk) [inline src/main/java/com/artipie/rpm/meta/XmlFile.java:61]: @ammaratef45 we need to create an `envelope` for `XMLStreamWriter` interface. We can call it `XmlWriterWrap`. Then `XmlFile` class should extend `XmlWriterWrap` and provide additional `alterTag` method.

File: src/main/java/com/artipie/rpm/meta/XmlFile.java [hunk 3]
```diff
@@ -157,4 +147,83 @@ private static XMLStreamWriter xmlStreamWriter(final Path path) {
             throw new UncheckedIOException("Failed to open file stream", err);
         }
     }
+
+    @Override
+    public void writeStartElement(
```
Comment (Vatavuk) [inline src/main/java/com/artipie/rpm/meta/XmlFile.java:152]: @ammaratef45 these method should be part of `XmlWriterWrap`. The envelope class should take `XMLStreamWriter` in ctor

**Comment by Vatavuk** [comment]
@ammaratef45 ping

File: src/main/java/com/artipie/rpm/meta/XmlWriter.java (hunk 1 above)
Comment (ammaratef45) [reply src/main/java/com/artipie/rpm/meta/XmlWriter.java:33]: @Vatavuk XMLStreamWriter has too many methods that we won't use, that will make the class so big

**Comment by ammaratef45** [comment]
@Vatavuk sorry for the late response, I've made some changes, please check them.

File: src/main/java/com/artipie/rpm/meta/XmlWriterWrap.java [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+        // Not implemeneted.
+    }
+
+    @Override
+    public void writeAttribute(
+        final String namespaceuri,
+        final String localname,
+        final String value
+    ) throws XMLStreamException {
+        // Not implemeneted.
```
Comment (Vatavuk) [inline src/main/java/com/artipie/rpm/meta/XmlWriterWrap.java:83]: @ammaratef45 please update all the unimplemented methods, its not a big effort we just need to delegate methods.

Comment (ammaratef45) [reply src/main/java/com/artipie/rpm/meta/XmlWriterWrap.java]: @Vatavuk check 5492a61

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**